nifi-extension-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/main/java/org/apache/nifi/kafka/processors/producer/key/AttributeKeyFactory.java

```diff
@@ -35,7 +36,8 @@ public AttributeKeyFactory(final String keyAttribute,
 
     @Override
     public byte[] getKey(final Map<String, String> attributes, final Record record) throws UnsupportedEncodingException {
-        final String keyAttributeValue = (keyAttribute == null) ? null : attributes.get(keyAttribute);
+        final String keyAttributeEffective = (keyAttribute == null) ? KafkaFlowFileAttribute.KAFKA_KEY : keyAttribute;
```

Comment:
For clarity, it looks like this logic could be handled in the constructor, instead of on every method invocation.

Reply:
Thanks; sure.  Believe it or not, I went back and forth about which style was better :)